challenge_02/base_solution.py

```diff
@@ -7,17 +10,27 @@ class SelfAttention(nn.Module):
     One of the most popular part in NLP.
     If you understand self attention, you will understand the heard of transformers.
     """
-    def __init__(self):
+    def __init__(self, embedding_length: int) -> None:
         super().__init__()
-        self.W = None
+        self.e_length = embedding_length
+        self.TW = self._trainable_weights()
+
+    def _trainable_weights(self) -> Tuple[nn.Module, nn.Module, nn.Module]:
+        q = nn.Linear(self.e_length, self.e_length, bias=False)
+        k = nn.Linear(self.e_length, self.e_length, bias=False)
+        v = nn.Linear(self.e_length, self.e_length, bias=False)
+        return q, k, v
```

Comment:
dá se to udělat v jednom nn.Linear

Reply:
dakujem znie to zaujímavo pozriem na to :) ale pre tento prípad by som to nechal ako je len kvôli prehladnosti